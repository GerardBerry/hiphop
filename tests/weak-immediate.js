"use hopscript"

var hh = require("hiphop");

var m =
    <hh.module>
      <hh.inputsignal name="S"/>
      <hh.outputsignal name="O"/>
      <hh.outputsignal name="F"/>
      <hh.outputsignal name="W"/>
      <hh.weakabort immediate signal_name="S">
	<hh.loop>
	  <hh.emit signal_name="O"/>
	  <hh.pause/>
	  <hh.emit signal_name="W"/>
	</hh.loop>
      </hh.weakabort>
      <hh.emit signal_name="F"/>
    </hh.module>

exports.prg = new hh.ReactiveMachine(m, "wabortimmediate")
