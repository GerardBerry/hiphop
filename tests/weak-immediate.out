wabortimmediate> S;
--- Output: O F 
wabortimmediate> ;
--- Output: 
wabortimmediate> !reset;
--- Automaton wabortimmediate reset
wabortimmediate> ;
--- Output: O 
wabortimmediate> S;
--- Output: O F W 
wabortimmediate> ;
--- Output: 
